File: 2019ACMRegionals/b.py
import sys

def diff(r,m):
    s = 2*max(r,m)
    pasta = [r]
    check = [False for i in range(s)]
    inc = 1 + (r == 1)
    a = r
    count = 1

    check[a] = True
    while not check[m]:
        #print("a: {}".format(a))
        #print("inc: {}".format(inc))
        check[inc] = True
        a = a + inc
        if a < s:
            check[a] = True
        count = count + 1
        for i in pasta:
            if a - i < s:
                check[a - i] = True
        pasta.append(a)
        while inc < s and check[inc]:
            inc = inc + 1
    return count

for s in sys.stdin:
    s = s.split()
    r = int(s[0])
    m = int(s[1])
    print(diff(r,m))

File: 2019ACMRegionals/b.cc
#include<algorithm>
#include<iostream>
#include<stdio.h>
#include<stdlib.h>
#include<vector>

using namespace std;

int main(void){
    long long int r, m, s, count, inc, a;

    while (cin >> r >> m){
        s = 2*max(r,m);
        vector<bool> check(s, false);
        vector<long long int> pasta;
        
        pasta.push_back(r);
        inc = 1 + (r == 1);
        a = r;
        count = 1;

        check[a] = true;
        while (!check[m]){
            check[inc] = true;
            a += inc;
            
            if (a < s)
                check[a] = true;
            for (size_t i=0; i<pasta.size(); i++)
                if (a - pasta[i] < s)
                    check[a-pasta[i]] = true; 
            pasta.push_back(a);
            while (inc < s && check[inc])
                inc++;
            count++;
        }
        cout << count << endl;
    }
    return 0;
}
